tests/storage_test.go

```diff
@@ -95,27 +96,40 @@ var _ = Describe("Storage", func() {
 			if nfsInitialized {
 				By("Deleting the VMI")
 				Expect(virtClient.VirtualMachineInstance(vmi.Namespace).Delete(vmi.Name, &metav1.DeleteOptions{})).To(Succeed())
-
 				By("Waiting for VMI to disappear")
 				tests.WaitForVirtualMachineToDisappearWithTimeout(vmi, 120)
-				// PVs can't be reused
-				tests.DeletePvAndPvc(_pvName)
+			}
+		})
 
+		AfterEach(func() {
+			if nfsInitialized {
 				By("Deleting NFS pod")
 				Expect(virtClient.CoreV1().Pods(tests.NamespaceTestDefault).Delete(tests.NFSTargetName, &metav1.DeleteOptions{})).To(Succeed())
+
 				By("Waiting for NFS pod to disappear")
 				tests.WaitForPodToDisappearWithTimeout(tests.NFSTargetName, 120)
 			}
 		})
+
+		AfterEach(func() {
+			if nfsInitialized {
+				// PVs can't be reused
+				By("Deleting PV and PVC")
+				tests.DeletePvAndPvc(pvName)
+			}
+		})
+
 		Context("[rfe_id:3106][crit:medium][vendor:cnv-qe@redhat.com][level:component]with Alpine PVC", func() {
-			table.DescribeTable("should be successfully started", func(newVMI VMICreationFunc, storageEngine string) {
+			table.DescribeTable("should be successfully started", func(newVMI VMICreationFunc, storageEngine string, family k8sv1.IPFamily) {
 				tests.SkipPVCTestIfRunnigOnKindInfra()
+				if family == k8sv1.IPv6Protocol {
+					libnet.SkipIpv6TestsOnSingleStackCluster(virtClient)
+				}
 
 				var ignoreWarnings bool
-				var pvName string
 				// Start the VirtualMachineInstance with the PVC attached
 				if storageEngine == "nfs" {
-					pvName = initNFS()
+					initNFS(family)
```

Comment:
I don't think this change is a positive direction, I don't like having initNFS as a function that when called, has the side effect of setting a global(ish) pvName.

Reply:
I will change it and the one in the migration_test to return the pv_name.